### PR TITLE
Migrate the python script to updatecli to update k3sroot

### PR DESCRIPTION
**Pull request by manuelbuil**
This PR creates a policy to update the `K3S_ROOT_VERSION` value in both Dockerfile and Makefile when there is a new release of the project. It uses an updatecli policy to do so. 

It also removes the script we were using right now to do the same thing

### CODE REVIEW

**Review by olblak — APPROVED** [review]

**Review by olblak — CHANGES_REQUESTED** [review]
the scm is mention values that don't exist in https://github.com/rancher/image-build-flannel/blob/master/updatecli/values.yaml

**Comment by olblak** [comment]
> the scm is mention values that don't exist in https://github.com/rancher/image-build-flannel/blob/master/updatecli/values.yaml

an example of values could be https://github.com/olblak/rancherlabs-policies/blob/main/updatecli/policies/flannel/default/values.yaml

File: updatecli/updatecli.d/updatek3sroot.yaml
Comment (macedogm) [inline updatecli/updatecli.d/updatek3sroot.yaml:53]: Agree with Olivier's comment. The default values here aren't specified in https://github.com/rancher/image-build-flannel/blob/master/updatecli/values.yaml. You can see an example in the repo's default Updatecli file:

https://github.com/rancher/image-build-flannel/blob/b5f8809a25d349a6ef3ca12d9c4d9ba1584b2285/updatecli/updatecli.d/updatecli.yml#L15-L18

File: updatecli/updatecli.d/updatek3sroot.yaml
Comment (macedogm) [inline updatecli/updatecli.d/updatek3sroot.yaml:31]: I don't remember if this will work out of the box or not, because the value in:
https://github.com/rancher/image-build-flannel/blob/b5f8809a25d349a6ef3ca12d9c4d9ba1584b2285/Dockerfile#L15

has the `"` around. Just commenting, because I really don't remember if the update will happen as expected. If you already tested this locally, please ignore.

Comment (olblak) [reply updatecli/updatecli.d/updatek3sroot.yaml:31]: In the current state, the update will drop the quote around.
I don't think it will be an issue but otherwise we can explicitely add them

File: updatecli/updatecli.d/updatek3sroot.yaml
Comment (manuelbuil) [reply updatecli/updatecli.d/updatek3sroot.yaml:53]: Given that they are not in the `values.yaml` and updatecli is supposed to be working, do I need the ones that are not in the values:
```
owner: '{{ .scm.default.owner }}'
repository: '{{ .scm.default.repository }}'
branch: '{{ .scm.default.branch }}'
```
? I wonder if I should add them or drop them from the policy

Comment (macedogm) [reply updatecli/updatecli.d/updatek3sroot.yaml:53]: They are specified directly in the current policy file (the only policy available at the moment in this repo): https://github.com/rancher/image-build-flannel/blob/b5f8809a25d349a6ef3ca12d9c4d9ba1584b2285/updatecli/updatecli.d/updatecli.yml#L19-L21

You could migrate those values from `updatecli/updatecli.d/updatecli.yml` to https://github.com/rancher/image-build-flannel/blob/master/updatecli/values.yaml and use them in your policy.

Comment (macedogm) [reply updatecli/updatecli.d/updatek3sroot.yaml:53]: Besides this, the rest LGTM.

**Review by macedogm — CHANGES_REQUESTED** [review]

File: updatecli/updatecli.d/updatek3sroot.yaml
Comment (macedogm) [inline updatecli/updatecli.d/updatek3sroot.yaml:53]:
```suggestion
      username: '{{ .github.username }}'
      user: '{{ requiredEnv "GITHUB_ACTOR"}}'
      email: '{{ .github.email }}'
      owner: '{{ .github.owner }}'
      repository: '{{ .github.repo }}'
      branch: '{{ .github.branch }}'
```

File: updatecli/updatecli.d/updatek3sroot.yaml
Comment (manuelbuil) [reply updatecli/updatecli.d/updatek3sroot.yaml]: thanks!

File: updatecli/updatecli.d/updatek3sroot.yaml
Comment (manuelbuil) [reply updatecli/updatecli.d/updatek3sroot.yaml]: done, thanks!

**Review by macedogm — APPROVED** [review]

**Review by rbrtbnfgl — APPROVED** [review]

File: updatecli/updatecli.d/updatek3sroot.yaml
Comment (rbrtbnfgl) [inline updatecli/updatecli.d/updatek3sroot.yaml:63]: Could we add here status/autogenerated?

**Review by rbrtbnfgl — CHANGES_REQUESTED** [review]

**Comment by manuelbuil** [comment]
Once CI passes I'll merge. I want to see if it works tomorrow :)

**Comment by manuelbuil** [comment]
I need to wait for Roberto's approval. It'll need to wait until Friday then

File: updatecli/updatecli.d/updatek3sroot.yaml
Comment (olblak) [reply updatecli/updatecli.d/updatek3sroot.yaml:63]: @rbrtbnfgl Do you man adding a label "status/auto-created" ?

Comment (rbrtbnfgl) [reply updatecli/updatecli.d/updatek3sroot.yaml:63]: We are already using that label on the PR generated with the github actions.

Comment (manuelbuil) [reply updatecli/updatecli.d/updatek3sroot.yaml:63]: I did it already @rbrtbnfgl , could you review please? :)

File: updatecli/updatecli.d/updatek3sroot.yaml
Comment (rbrtbnfgl) [inline updatecli/updatecli.d/updatek3sroot.yaml:11]: I think this is not going to work the env defined on the updatecli action is GH_TOKEN

Comment (manuelbuil) [reply updatecli/updatecli.d/updatek3sroot.yaml:11]: you are right, I'm dumb. Fixed!

Comment (olblak) [reply updatecli/updatecli.d/updatek3sroot.yaml:11]: Good catch

According 
https://github.com/rancher/image-build-flannel/blob/b5f8809a25d349a6ef3ca12d9c4d9ba1584b2285/.github/workflows/updatecli.yml#L55

It should be `UPDATECLI_GITHUB_TOKEN`

Comment (olblak) [reply updatecli/updatecli.d/updatek3sroot.yaml:11]:
```suggestion
     token: '{{ requiredEnv "UPDATECLI_GITHUB_TOKEN" }}'
```

**Comment by manuelbuil** [comment]
For $REASONS I don't understand yet, token and user can't be passed as values parameters. This does not work in the policy:
```
token: '{{ .github.token }}'
````
whereas this works:
```
token: '{{ requiredEnv "UPDATECLI_GITHUB_TOKEN" }}'
```

I updated the policy with this change

**Comment by rbrtbnfgl** [comment]
From the PR it removes the " from the argument on the Dockerfile  is it right?
Is it possible to squash all the commit before making the PR from updatecli?

**Comment by manuelbuil** [comment]
> From the PR it removes the " from the argument on the Dockerfile is it right? Is it possible to squash all the commit before making the PR from updatecli?

re: the `"`, yes, it was discussed in one of the comments. It is apparently impossible to leave it but according to my tests, it does not matter, the docker build still works.

re: the squash, I don't know. Maybe @olblak or @macedogm know

**Comment by olblak** [comment]
> re: the ", yes, it was discussed in one of the comments. It is apparently impossible to leave it but according to my tests, it does not matter, the docker build still works.
If it's important there is a way to add prefix and suffix with something like 

```
  dockerfile:
    name: "Bump to latest k3s-root version in Dockerfile"
    kind: dockerfile
    scmid: default
    sourceid: k3sroot
    transformers:
      - addprefix: '"'
      - addsuffix: '"'
    spec:
      file: "Dockerfile"
      instruction:
        keyword: "ARG"
        matcher: "K3S_ROOT_VERSION"
```

Bump it makes the manifest more complex without adding real value

**Comment by olblak** [comment]
> For $REASONS I don't understand yet, token and user can't be passed as values parameters. This does not work in the policy:
> token: '{{ .github.token }}'
> whereas this works:
> token: '{{ requiredEnv "UPDATECLI_GITHUB_TOKEN" }}'
> I updated the policy with this change and got it working ==> https://github.com/rancher/image-build-flannel/pull/53

> token: '{{ .github.token }}'

Means you want to retrieve the token from the values or a secret file.
Storing token in an unencrypted values file is not a good practice.
Another option is to use a SOPS file to store the token but then you would need additional configuration to unencrypt the secret file from GitHub action

> token: '{{ requiredEnv "UPDATECLI_GITHUB_TOKEN" }}'

Means you retrieve the token from the environment variable "UPDATECLI_GITHUB_TOKEN"

**Review by rbrtbnfgl — APPROVED** [review]

**Comment by matttrach** [comment]
You can also get the name of the environment variable from values: 
```
token: '{{ requiredEnv .github.token }}'
```
where in the values.yml you have:
```
github:
  token: UPDATECLI_GITHUB_TOKEN
```
This helps make the manifest a little more generic.

**Pull request merged**
